src/test/rustdoc-ui/intra-doc/field-ice.stderr

```diff
@@ -0,0 +1,15 @@
+error: incompatible link kind for `Foo::bar`
+  --> $DIR/field-ice.rs:5:6
+   |
+LL | /// [`Foo::bar()`] 
+   |      ^^^^^^^^^^^^ help: to link to the field, remove the disambiguator: ``Foo::bar``
```

Comment:
The message is really not obvious. I think we should create a new error type so you can generate an error message looking like this:

```
help: to link to the field, remove the parenthesis: `Foo::bar`
```

(without the extra backquotes too)

Reply:
the prefix and suffix are all mapped to Disambiguator::Kind(DefKind::Fn):

`fn@foo`, `function@foo`, `method@foo`, and `foo()`.

There are two ways to approach this, either we only allow one kind of those four to exist, or we need to refactor to include that information inside `Disambiguator`.

Reply:
> either we only allow one kind of those four to exist

This is a breaking change, it's not allowed